src/controllers/users.go

```diff
@@ -0,0 +1,169 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"github.com/dana-team/platform-backend/src/types"
+	"go.uber.org/zap"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+)
+
+type UserController interface {
+	// GetUsers get  users from specified namespace and returns them as users.
+	GetUsers(namespace string) (types.UsersOutput, error)
+
+	// GetUser get specific user from specified namespace and returns it as user.
+	GetUser(userIdentifier types.UserIdentifier) (types.User, error)
+
+	// AddUser creates a new roleBinding in the specified namespace.
+	AddUser(user types.UserInput) (types.User, error)
+
+	// DeleteUser deletes user in the specified namespace.
+	DeleteUser(userIdentifier types.UserIdentifier) (types.DeleteUserResponse, error)
+
+	// UpdateUser updates user in the specified namespace.
+	UpdateUser(user types.UserInput) (types.User, error)
+}
+
+type userController struct {
+	client kubernetes.Interface
+	ctx    context.Context
+	logger *zap.Logger
+}
+
+func NewUserController(client kubernetes.Interface, context context.Context, logger *zap.Logger) UserController {
+	return &userController{
+		logger: logger,
+		client: client,
+		ctx:    context,
+	}
+}
+
+func (u *userController) GetUsers(namespace string) (types.UsersOutput, error) {
+	userOutputs := types.UsersOutput{}
+	u.logger.Debug(fmt.Sprintf("Trying to get all rolebindings in %q namespace", namespace))
+
+	roleBindings, err := u.client.RbacV1().RoleBindings(namespace).List(u.ctx, metav1.ListOptions{})
+	if err != nil {
+		u.logger.Error(fmt.Sprintf("Could not get rolebindings with error: %v", err.Error()))
+		return types.UsersOutput{}, err
+	}
+	u.logger.Debug("Fetched all rolebindings successfully")
+
+	for _, roleBinding := range roleBindings.Items {
+		userOutputs.Users = append(userOutputs.Users, types.User{Name: roleBinding.Name, Role: convertToPlatformRole(roleBinding.RoleRef.Name)})
+	}
+	userOutputs.Count = len(roleBindings.Items)
+
+	return userOutputs, nil
+}
+
+func (u *userController) GetUser(userIdentifier types.UserIdentifier) (types.User, error) {
+	userOutput := types.User{}
+	u.logger.Debug(fmt.Sprintf("Trying to fetch rolebinding %q in %q namespace", userIdentifier.UserName, userIdentifier.NamespaceName))
+
+	roleBinding, err := u.client.RbacV1().RoleBindings(userIdentifier.NamespaceName).Get(u.ctx, userIdentifier.UserName, metav1.GetOptions{})
+	if err != nil {
+		u.logger.Error(fmt.Sprintf("Could fetch rolebinding %q with error: %v", userIdentifier.UserName, err.Error()))
+		return userOutput, err
+	}
+	u.logger.Debug(fmt.Sprintf("fetched roleBinding %q successfully", userIdentifier.UserName))
+	userOutput.Name = roleBinding.Name
```

Comment:
nit: Initialize these variables inside `types.User{ ... }`, e.g. an example of it:
```
user := types.User {
   Name: roleBinding.Name,
   Role: convertToPlatformRole(roleBinding.RoleRef.Name)
}
```

Reply:
when i debug i prefer to watch a variable and the way i did it make it more comfortable to debug for me